docs/DeveloperGuide.md

```diff
@@ -320,6 +320,83 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**Use case UCV01: Create a volunteer**
+
+**MSS**
+
+1.  User requests to create a volunteer.
+2.  iVolunteer shows the volunteer created and appends the volunteer to the end of the volunteer list.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. Invalid Command Word.
+    * 1a1. iVolunteer prompts Volunteer Coordinator to provide a valid command. 
+
+    Use case resumes from step 1.
+* 1b. Missing arguments for mandatory fields.
+    * 1b1. System prompts Volunteer Coordinator to provide arguments for all mandatory fields.
+
+    Use case resumes from step 1.
+* 1c. Parameters are not separated by a single space.
+    * 1c1. System prompts Volunteer Coordinator to separate parameters with a single space.
+
+    Use case resumes from step 1.
+* 1d. Invalid email.
+    * 1d1. System prompts Volunteer Coordinator to use the correct email format.
+
+    Use case resumes from step 1.
+* 1e. Invalid phone number.
+    * 1e1. System prompts Volunteer Coordinator to use a valid 8-digit phone number.
+
+    Use case resumes from step 1.
+* 1f. Name exceeds 30 characters, or name is empty.
+    * 1f1. System prompts Volunteer Coordinator to use a volunteer name between 1-30 characters.
+
+    Use case resumes from step 1.
+
+**Use case UCV02: List all volunteers**
+
+**MSS**
+
+1.  User requests to list all volunteers.
+2.  iVolunteer shows a list of all volunteers.
+
+    Use case ends.
+
+**Use case UCV03: Delete all volunteers**
+
+**MSS**
+
+1.  User <u>lists all volunteers (UCV02)</u>.
+2.  User requests to delete a specific volunteer in the volunteer list.
+3.  iVolunteer removes the volunteer from all events he/she is in.
+4.  iVolunteer deletes the volunteer.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty.
+
+  Use case ends.
+
+* 2a. The given index is invalid.
+
+    * 2a1. iVolunteer shows an error message that there is no such volunteer in the given index.
+
+      Use case resumes from step 2.
+
+**Use case UCV04: Clear all volunteers in volunteer list**
+
+**MSS**
+
+1. User clears all volunteers in volunteer list.
+2. The volunteer list becomes empty.
+
+    Use case ends.
```

Comment:
Do we really need a feature to delete all volunteers? Feels like won't be necessary

Reply:
this one is analogous to `clear` in AB3 - clears volunteers **from view** but doesn't really delete any volunteer

Reply:
or else the `vlist` a bit no point HAHA

Reply:
oki